src/pages/iou/steps/IOUAmountPage.js

```diff
@@ -183,6 +185,7 @@ class IOUAmountPage extends React.Component {
                                 ref={el => this.textInput = el}
                                 value={this.state.amount}
                                 placeholder="0"
+                                maxLength={CONST.IOU.AMOUNT_MAX_LENGTH}
```

Comment:
Why are we passing this as a prop to `TextInputAutoWidth`? It doesn't seem like we will allow the setting a value that is larger anyway?

Reply:
Good point. I added the prop first and then the number pad change. Will remove.

Reply:
@marcaaron I checked again there are two functions `updateAmount` and `updateAmountNumberPad`.  

We can do one of the three things:
1. Retain my `maxLength` change
2. Add the length change in `updateAmount` also
3. Add the length change in `updateAmount` and call it inside `updateAmountNumberPad` instead of setting the state.

Please let me know which one suits better?

Reply:
For now, I've kept 1. and updated the PR.

Reply:
I would lean towards including it in the `validateAmount()` method (which both of these methods you mentioned use) and prevent setting state instead of using the `maxLength` on the `TextInput`. It's a controlled component so if we prevent setting state the value should not change?

Reply:
Okay, that worked. PR updated.